back/app/services/authentication_service.rb

```diff
@@ -42,7 +42,7 @@ def prevent_user_account_hijacking(user)
     return nil if !user
 
     if user.confirmation_required? && !user.email_confirmed_at && user.password_digest
-      user.destroy!
+      DeleteUserJob.perform_now(user)
```

Comment:
@jamesspeake I also used the job here, as discussed in today's meeting. It shouldn't change anything apart from logging `Activity`.